core/src/main/scala/kafka/BrokerStart.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka;
+
+import java.util.Arrays;
+import java.util.Properties;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import kafka.metrics.KafkaMetricsReporter$;
+import kafka.server.BrokerServerStartable;
+import kafka.server.KafkaConfig;
+import kafka.utils.CommandLineUtils;
+import kafka.utils.VerifiableProperties;
+import org.apache.kafka.common.utils.Utils;
+import kafka.server.KafkaServer;
+
+/**
+ * Main class that starts a Kafka broker and any associated threads
+ * This class is similar to Kafka.scala
+ */
+public class BrokerStart  {
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Properties serverProps = Kafka$.MODULE$.getPropsFromArgs(args);
```

Comment:
Do you really need to add `MODULE$` here? Scala tries to add static forwarder methods so that you can call such methods from Java as normal (ie `Kafka.getPropsFromArgs(args)`). There are cases where it can't do it due to name clashes, but it's worth trying (and make sure you compile the code, IDEA may misreport an error).